ivy/functional/ivy/data_type.py

```diff
@@ -580,11 +581,10 @@ def can_cast(
     """
     if isinstance(from_, (ivy.Array, ivy.NativeArray)):
         from_ = from_.dtype
-    try:
-        dtype = ivy.promote_types(from_, to)
-        return dtype == to
-    except KeyError:
-        return False
+    dtype = ivy.promote_types(from_, to)
+    if dtype == to:
```

Comment:
let's instead do a `return dtype == to` here, as I suppose the additional `return False` in the `except` isn't needed anymore

Reply:
Done